launcher/src/main/java/org/wso2/carbon/launcher/extensions/DropinsBundleDeployerUtils.java

```diff
@@ -0,0 +1,273 @@
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.wso2.carbon.launcher.extensions;
+
+import org.wso2.carbon.launcher.Constants;
+import org.wso2.carbon.launcher.extensions.model.BundleInfo;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.jar.JarFile;
+import java.util.jar.Manifest;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * This class defines the utility functions used when implementing the dropins deployment capability.
+ *
+ * @since 5.1.0
+ */
+public class DropinsBundleDeployerUtils {
+    private static final Logger logger = Logger.getLogger(DropinsBundleDeployerUtils.class.getName());
+    private static final String dropinsDirectory = "dropins";
+
+    private static List<BundleInfo> newBundlesInfo;
+
+    /**
+     * Updates the specified Carbon profile's bundles.info file based on the OSGi bundles deployed in the dropins
+     * directory. The OSGi bundle information in the bundles.info file in a Carbon profile is used to install and
+     * start the bundles at the server startup for the particular profile.
+     * <p>
+     * The mechanism used in updating the bundles.info file is as follows:
+     * 1. The new OSGi bundle information from the bundles currently existing within the dropins folder are obtained.
+     * The new OSGi bundle information are read only once for updating one or more Carbon profiles.
+     * 2. The existing OSGi dropins bundle information are compared with the newly retrieved bundle information and
+     * the bundles.info file is updated only if the new bundle information are different from the existing.
+     * 3. The new OSGi bundle information replace the existing dropins bundle information from the bundles.info file.
+     * The OSGi bundle information of the non-dropins bundles, retrieved from the bundles.info file and the new dropins
+     * OSGi bundle information are merged together.
+     * 4. Updates the bundles.info file with the OSGi bundle information retrieved in step 3.
+     *
+     * @param carbonHome    the {@link String} representation of carbon.home
+     * @param carbonProfile the bundles.info file to be updated
+     * @throws IOException if an I/O error occurs
+     */
+    public static synchronized void executeDropinsCapability(String carbonHome, String carbonProfile)
+            throws IOException {
+        Path dropinsDirectoryPath = Paths.get(carbonHome, Constants.OSGI_REPOSITORY, dropinsDirectory);
+        Path bundlesInfoFile = Paths.
+                get(carbonHome, Constants.OSGI_REPOSITORY, Constants.PROFILE_PATH, carbonProfile, "configuration",
+                        "org.eclipse.equinox.simpleconfigurator", "bundles.info");
```

Comment:
It would be better to move "bundles.info" to a Constant and reused (including Tests).